src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs

```diff
@@ -0,0 +1,276 @@
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetReference_NonInvoked()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void Bar() { Action<int> i = GetFoo; } }",
+@"using System; class C { int Foo { get { } } void Bar() { Action<int> i = {|Conflict:Foo|}; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetReference_ImplicitReference()
+        {
+            Test(
+@"using System.Collections; class C { public IEnumerator [||]GetEnumerator() { } void Bar() { foreach (var x in this) { } } }",
+@"using System.Collections; class C { public IEnumerator Enumerator { get { } } void Bar() { {|Conflict:foreach (var x in this) { }|} } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { } }",
+@"using System; class C { int Foo { get { } set { } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSetReference_NonInvoked()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { } void Bar() { Action<int> i = SetFoo; } }",
+@"using System; class C { int Foo { get { } set { } } void Bar() { Action<int> i = {|Conflict:Foo|}; } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_SetterAccessibility()
+        {
+            Test(
+@"using System; class C { public int [||]GetFoo() { } private void SetFoo(int i) { } }",
+@"using System; class C { public int Foo { get { } private set { } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_ExpressionBodies()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() => 0; void SetFoo(int i) => Bar(); }",
+@"using System; class C { int Foo { get { return 0; } set { Bar(); } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_GetInSetReference()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { } void Bar() { SetFoo(GetFoo() + 1); } }",
+@"using System; class C { int Foo { get { } set { } } void Bar() { Foo = Foo + 1; } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_UpdateSetPrameterName()
+        {
+            Test(
```

Comment:
Having re-read this I realized I asked the wrong question, but thanks for humoring me and answering anyways.

The question I really wanted to ask was: could you add a test for a local variable in the body named `value` and the rename conflict that will arise.